beps/0001-notifications-system/README.md

```diff
@@ -160,89 +166,194 @@ The following backend service interfaces are added as part of this proposal.
 ```ts
 // TODO - We may want to add an additional wrapping here with interfaces for Notification and NotificationParameters
 
-interface SendNotificationRequest {
-  receivers: SignalReceivers;
+export type NotificationRecipients =
+  | {
+      type: 'entity';
+      entityRef: string | string[]; // typically a user or group entity, otherwise `spec.owner` of the entity/entities
+    }
+  | {
+      type: 'broadcast'; // all users
+    };
 
-  /* TODO - please contribute :) */
-}
+export type NotificationType = 'undone' | 'done' | 'saved';
 
-interface SendNotificationResponse {
+export type Notification = {
   id: string;
-  receivers: SignalReceivers;
+  userRef: string;
+  title: string;
+  description: string;
+  link: string;
+  created: Date;
+  saved: boolean;
+  read?: Date;
+  topic?: string;
+  updated?: Date;
+};
 
-  /* TODO - please contribute :) */
+interface SendNotificationRequest {
+  recipients: NotificationRecipients;
+  title: string;
+  description: string;
+  link: string;
+  topic?: string;
 }
 
 interface NotificationService {
-  sendNotification(
-    request: SendNotificationRequest,
-  ): Promise<SendNotificationResponse>;
+  sendNotification(request: SendNotificationRequest): Promise<Notification[]>;
 }
 ```
 
+Notification is always targeted to a single user unless it's a broadcast. The `notification-backend` will figure out which users will receive notification based on the `recipients` parameter.
+
+Each notification contains a `title`, a `description` and a `link` for further information. The `created`, `id`, `read` and `saved` properties are handled by the backend based and cannot be changed during the notification creation.
+
+Calling `sendNotification` should never throw an error so that it doesn't block the current processing. Notifications should be considered as second-level citizens that are not critical if not delivered.
+
+The `notification-backend` shall provide necessary parameters for paging and filtering notifications from the frontend.
+
 #### `SignalService`
 
-```ts
-type SignalReceivers =
-  | {
-      type: 'entity';
-      entityRef: string; // typically a user or group entity
-    }
-  | {
-      type: 'broadcast'; // all users
-    };
+Upon creating a new notification, a signal will be emitted by the `notification-backend`.
+
+The channel for notifications will be named `notifications` and the message will contain necessary information about the notification for rendering as well as action telling to refresh the notifications using the REST API.
+
+The notification payload must be in the signal to be able to show notification in the UI immediately after receiving the signal.
```

Comment:
Initially we said this was not the case, but tbh it's prolly worth reconsidering. In the end it should cause less traffic to the backend overall if we don't need the followup request to the notifications backend after receiving a signal.

Reply:
To be able to show the notifications easily using the Web Notification API for example, we should include the data to payload. How to handle the data change in frontend is another topic. But I would leave this here for now, let's discuss a bit more later